### ISSUE TITLE
Run clang-format

### ISSUE THREAD

**Issue by drwells** [IBAMR/IBAMR#487]
I just ran clang-format and it wants to reindent essentially every single file we have (about 450 files). What version of clang-format did we use in the past? I tried both 6 and 8 and got huge change sets.

In the future, we should set up a travis instance to check indentation so that we cannot commit to master unless someone has indented their PR with clang-format. We should also run clang-format over everything (in C++11 mode) after we make the next release.

**Comment by boyceg** [comment]
I currently have:
```
$ clang-format --version
clang-format version 8.0.0 (tags/google/stable/2018-08-24)
```
There doesn’t seem to be a way to force different versions to interpret the settings consistently, so there can be drift.

I use a pre-commit hook that runs clang-format on the diffs, and I think that we explicitly encourage folks to do this.

There is a script that you can use to reformat the library in IBAMR/scripts/reformat.sh. Here is what it does:
```
#! /bin/bash
FILES="`find examples include ibtk/include ibtk/examples ibtk/src src tests -name *.cpp -o -name *.h`"
for file in $FILES; do
  echo $file
  clang-format -i -style=file $file
done
```
When I do this, I see a lot of formatting changes in the examples. In the main source code, I see changes mostly related to line breaks for strings. It also seems like the indentation rules have changed about `extern` --- all of the FORTRAN prototypes are getting reindented for some reason.

**Comment by drwells** [comment]
I saw the same thing (prototypes). I also saw that lines where we added `override` need to be split now.

**Comment by drwells** [comment]
Fixed by #599.
